TestTimeConversation fails: conv_test.go:52: strptime("2016 100 PM 11", "%Y %j %p %I"): got "2015-12-31T11:00:00Z", expected "2016-04-09T23:00:00Z"

Happened pretty randomly: 
```
--- FAIL: TestTimeConversion (0.00s)
       	conv_test.go:52: strptime("2016 100 PM 11", "%Y %j %p %I"): got "2015-12-31T11:00:00Z", expected "2016-04-09T23:00:00Z"
FAIL
FAIL   	github.com/cockroachdb/cockroach/util/timeutil 	0.416s
ok     	github.com/cockroachdb/cockroach/util/tracing  	0.023s
?      	github.com/cockroachdb/cockroach/util/uuid     	[no test files]
make: *** [test] Error 1
✘-2 ~/go/src/github.com/cockroachdb/cockroach [master|✚ 7…2⚑ 3]
00:14 $ date
Thu Oct  6 00:15:12 EDT 2016
```

Happened again:

```
--- FAIL: TestTimeConversion (0.00s)
       	conv_test.go:52: strptime("2016 100 PM 11", "%Y %j %p %I"): got "2015-12-31T11:00:00Z", expected "2016-04-09T23:00:00Z"

01:23 $ date
Thu Oct  6 01:23:42 EDT 2016
```

--- a/util/timeutil/conv_test.go
+++ b/util/timeutil/conv_test.go
@@ -34,7 +34,9 @@ func TestTimeConversion(t *testing.T) {
 		{`20061012 01:03:02`, `%Y%m%d %H:%S:%M`, `2006-10-12T01:02:03Z`, ``, ``},
 		{`2018 10 4`, `%Y %W %w`, `2018-03-08T00:00:00Z`, ``, ``},
 		{`2018 10 4`, `%Y %U %w`, `2018-03-15T00:00:00Z`, ``, ``},
-		{`2016 100 PM 11`, `%Y %j %p %I`, `2016-04-09T23:00:00Z`, ``, ``},
+		// %j cannot be used reliably to parse a date from text; so test
+		// it only for rendering.
+		{`20161012 PM 11`, `%Y%m%d %p %I`, `2016-10-12T23:00:00Z`, `%j`, `286`},
 		{`Wed Oct 05 2016`, `%a %b %d %Y`, `2016-10-05T00:00:00Z`, ``, ``},
 		{`Wednesday October 05 2016`, `%A %B %d %Y`, `2016-10-05T00:00:00Z`, ``, ``},
 	}

